src/main/java/spoon/pattern/PatternBuilder.java

```diff
@@ -112,11 +112,11 @@ public static PatternBuilder create(CtElement... elems) {
 	}
 
 	protected PatternBuilder(List<CtElement> template) {
-		this.templateTypeRef = getDeclaringTypeRef(template);
-		this.patternModel = Collections.unmodifiableList(new ArrayList<>(template));
 		if (template == null) {
 			throw new SpoonException("Cannot create a Pattern from an null model");
 		}
+		this.templateTypeRef = getDeclaringTypeRef(template);
```

Comment:
this is easily mergeable, what about putting it in a separate PR? 

(for changes that look quite different like this one, the discussion and merge is easier in separate PRs, thanks!)

Reply:
Would you like me to make 3 separate PR's for this issue again?